float64[] sphere_data
